File: TangramJava/src/poo4/Rect.java
package poo4;

import processing.core.PApplet;

class Rect extends Shape {
  protected float _edge;

  public Rect(PApplet parent) {
    super(parent);
    setEdge(100);
  }

  @Override
  protected void drawShape() {
    _parent.rectMode(PApplet.CENTER);
    _parent.rect(0, 0, edge(), edge());
  }

  public float edge() {
    return _edge;
  }

  public void setEdge(float e) {
    _edge = e;
  }
}

File: docs/js/rect.js
class Rect extends Shape {
  constructor () {
    super();
    this.edge = 100;
    //this.setEdge(100);  
  }

  drawShape() {
    rectMode(CENTER);
    rect(0, 0, 100, 100);
    //rect(0, 0, this.edge(), this.edge());
  }

  /*
  edge() {
    return this.edge;
  }

  setEdge(e) {
    this.edge = e;
  }
  */
}
